### ISSUE TITLE
Video recording size is too large and not mirror recordingg

### ISSUE THREAD

**Issue by LocalBaat** [cordova-plugin-camera-preview/cordova-plugin-camera-preview#648]
Recording is too large ever after giving small height and width and quality. And recorded video is not as mirror reflection

**Comment by ktangy63** [comment]
I have this same issue. Even though the width and height are set to a low resolution the video is recorded at 3840x2160. Quality does not seem to have any effect either.

**Comment by pocopuede** [comment]
Yes, the options passed to CameraPreview.startRecordVideo({options....}, ...) don't work in the plugin. After check in java files, I have found how to reduce or change the video size when is recorded.

In Cordova project:
In platforms > android > app > src > main > java > com > cordovaplugincamerapreview > CameraActivity.java

In line 683 I can find the method "startRecord", with the params passed from the javascript. In this method below I found the next code near of line 711

`      CamcorderProfile profile;
        if (CamcorderProfile.hasProfile(defaultCameraId, CamcorderProfile.QUALITY_HIGH)) {
          profile = CamcorderProfile.get(defaultCameraId, CamcorderProfile.QUALITY_HIGH);
        } else {
          if (CamcorderProfile.hasProfile(defaultCameraId, CamcorderProfile.QUALITY_480P)) {
            profile = CamcorderProfile.get(defaultCameraId, CamcorderProfile.QUALITY_480P);
          } else {
            if (CamcorderProfile.hasProfile(defaultCameraId, CamcorderProfile.QUALITY_720P)) {
              profile = CamcorderProfile.get(defaultCameraId, CamcorderProfile.QUALITY_720P);
            } else {
              if (CamcorderProfile.hasProfile(defaultCameraId, CamcorderProfile.QUALITY_1080P)) {
                profile = CamcorderProfile.get(defaultCameraId, CamcorderProfile.QUALITY_1080P);
              } else {
                profile = CamcorderProfile.get(defaultCameraId, CamcorderProfile.QUALITY_LOW);
              }
            }
          }
        }`

I have commented al this code and put over or below the next code:
`      CamcorderProfile profile;
        profile = CamcorderProfile.get(defaultCameraId, CamcorderProfile.QUALITY_480P);`

Changing the value of CamcorderProfile 'QUALITY_480P' with other options just like:
- QUALITY_HIGH
- QUALITY_480P
- QUALITY_720P
- QUALITY_1080P
- QUALITY_LOW

I get the recorded video with different sizes and qualities.

I hope this will be useful for you.

**Comment by ktangy63** [comment]
Thank you for the feedback on this feature. I have tried to replicate what you did and cannot get it to work in my project. I am new to this area and would really like to get this to work. Are you open to doing some consulting with me on this?

**Comment by ktangy63** [comment]
Sorry, my comments for each case are off above.

Also, I thought it would be useful to have the phone report all of the resolutions it supports so I added this function:

  private boolean getSupportedVideoSizes(CallbackContext callbackContext) {
    if (this.hasCamera(callbackContext) == false) {
      return true;
    }

    List<Camera.Size> supportedVideoSizes;
    Camera camera = fragment.getCamera();
    supportedVideoSizes = camera.getParameters().getSupportedVideoSizes();

    if (supportedVideoSizes != null) {
      JSONArray jsonVideoSizes = new JSONArray();
      for (int i=0; i<supportedVideoSizes.size(); i++) {
        Camera.Size size = supportedVideoSizes.get(i);
        jsonVideoSizes.put(size);
      }

      callbackContext.success(jsonVideoSizes);
      return true;
    }

    callbackContext.error("Camcorder profile access error");
    return true;
  }
}

It works but it returns cryptic information that I will need to figure out how to translate:

:: Requested videoProfile
app.js:82 :: android.hardware.Camera$Size@7ccac70, 
app.js:82 :: android.hardware.Camera$Size@77ccf70, 
app.js:82 :: android.hardware.Camera$Size@65d4c50, 
app.js:82 :: android.hardware.Camera$Size@53dc468, 
app.js:82 :: android.hardware.Camera$Size@53dc3c0, 
app.js:82 :: android.hardware.Camera$Size@50ddab8, 
app.js:82 :: android.hardware.Camera$Size@3fe4e00, 
app.js:82 :: android.hardware.Camera$Size@3be6920, 
app.js:82 :: android.hardware.Camera$Size@3be67b8, 
app.js:82 :: android.hardware.Camera$Size@31eacf0, 
app.js:82 :: android.hardware.Camera$Size@3be6740, 
app.js:82 :: android.hardware.Camera$Size@27ef0c0, 
app.js:82 :: android.hardware.Camera$Size@27ef000, 
app.js:82 :: android.hardware.Camera$Size@27eefd0, 
app.js:82 :: android.hardware.Camera$Size@1ff2700, 
app.js:82 :: android.hardware.Camera$Size@18f55b0, 
app.js:82 :: android.hardware.Camera$Size@18f5678, 
app.js:82 :: android.hardware.Camera$Size@18f5600, 
app.js:82 :: android.hardware.Camera$Size@1676730, 
app.js:82 :: android.hardware.Camera$Size@13f7860, 
app.js:82 :: android.hardware.Camera$Size@13f7810, 
app.js:82 :: android.hardware.Camera$Size@13f77e8, 
app.js:82 :: android.hardware.Camera$Size@afb580, 
app.js:82 :: android.hardware.Camera$Size@9fbc30, 
app.js:82 :: android.hardware.Camera$Size@57dac0,
